yarn-project/end-to-end/src/integration_l1_publisher.test.ts

```diff
@@ -122,46 +126,104 @@ describe('L1Publisher integration', () => {
     return makeEmptyProcessedTxFromHistoricTreeRoots(historicTreeRoots);
   };
 
-  const makeContractDeployProcessedTx = async (seed = 0x1) => {
-    const tx = await makeEmptyProcessedTx();
-    tx.data.end.newContracts = [makeNewContractData(seed + 0x1000)];
-    return tx;
-  };
+  const makeBloatedProcessedTx = async (seed = 0x1) => {
+    const publicTx = makePublicTx(seed);
+    const kernelOutput = KernelCircuitPublicInputs.empty();
+    kernelOutput.constants.historicTreeRoots = await getCombinedHistoricTreeRoots(builderDb);
+    kernelOutput.end.stateTransitions = range(STATE_TRANSITIONS_LENGTH, seed + 0x500).map(
+      i => new PublicDataTransition(fr(i), fr(0), fr(i + 10)),
+    );
+
+    const tx = await makeProcessedTx(publicTx, kernelOutput, makeProof());
 
-  const makePrivateProcessedTx = async (seed = 0x1) => {
-    const tx = await makeEmptyProcessedTx();
     tx.data.end.newCommitments = range(KERNEL_NEW_COMMITMENTS_LENGTH, seed + 0x100).map(fr);
     tx.data.end.newNullifiers = range(KERNEL_NEW_NULLIFIERS_LENGTH, seed + 0x200).map(fr);
+    tx.data.end.newL2ToL1Msgs = range(KERNEL_NEW_L2_TO_L1_MSGS_LENGTH, seed + 0x300).map(fr);
+    tx.data.end.newContracts = [makeNewContractData(seed + 0x1000)];
+
     return tx;
   };
 
-  const makePublicCallProcessedTx = async (seed = 0x1) => {
-    const publicTx = makePublicTx(seed);
-    const kernelOutput = KernelCircuitPublicInputs.empty();
-    kernelOutput.end.stateReads[0] = new PublicDataRead(fr(1), fr(0));
-    kernelOutput.end.stateTransitions[0] = new PublicDataTransition(fr(2), fr(0), fr(12));
-    kernelOutput.constants.historicTreeRoots = await getCombinedHistoricTreeRoots(builderDb);
-    return await makeProcessedTx(publicTx, kernelOutput, makeProof());
-  };
+  it(`Build ${numberOfConsecutiveBlocks} blocks of 4 bloated txs building on each other`, async () => {
+    const stateInRollup_ = await rollup.read.rollupStateHash();
+    expect(hexStringToBuffer(stateInRollup_.toString())).toEqual(Buffer.alloc(32, 0));
+
+    const blockNumber = await publicClient.getBlockNumber();
+
+    for (let i = 0; i < numberOfConsecutiveBlocks; i++) {
+      const l1ToL2Messages = range(NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP, 128 * i + 1 + 0x400).map(fr);
+      const txs = [
+        await makeBloatedProcessedTx(128 * i + 32),
+        await makeBloatedProcessedTx(128 * i + 64),
+        await makeBloatedProcessedTx(128 * i + 96),
+        await makeBloatedProcessedTx(128 * i + 128),
+      ];
+      const [block] = await builder.buildL2Block(1 + i, txs, l1ToL2Messages);
+
+      /*// Useful for sol tests block generation
+      const encoded = block.encode();
```

Comment:
this is lingering although its probably convienent to keep

Reply:
Kept it like this to make it easy for later use. The sol decoder test is there really only for us to more easily extend and test new decoders in foundry.